resolver-dns-native-macos/src/main/java/io/netty/resolver/dns/macos/MacOSDnsServerAddressStreamProvider.java

```diff
@@ -51,6 +52,16 @@
     private static final long REFRESH_INTERVAL = TimeUnit.SECONDS.toNanos(10);
 
     static {
+        // Preload all classes that will be used in the OnLoad(...) function of JNI to eliminate the possiblity of a
+        // class-loader deadlock. This is a workaround for https://github.com/netty/netty/issues/11209.
+
+        // This needs to match all the classes that are loaded via NETTY_JNI_UTIL_LOAD_CLASS or looked up via
+        // NETTY_JNI_UTIL_FIND_CLASS.
+        ClassInitializerUtil.tryLoadClasses(PlatformDependent.getClassLoader(MacOSDnsServerAddressStreamProvider.class),
+                // netty_resolver_dns_macos
+                byte[].class, String.class
+        );
```

Comment:
I don't think it is strictly necessary to load classes from the JDK ( the boot and platform loaders ), but will do no harm. It is probably best to follow a simple idiom - preload all classes referred to from within JNI_OnLoad <period> - rather than a more complex rule.

Reply:
Yeah that's what I did ... better safe than sorry